app/src/main/java/openfoodfacts/github/scrachx/openfood/views/HistoryScanActivity.java

```diff
@@ -290,4 +312,39 @@ public void onDraw(Canvas c, RecyclerView parent, RecyclerView.State state) {
         }
     }
 
+
+    @OnClick(R.id.scanFirst)
+    protected void onScanFirst() {
+
+
+        if (Utils.isHardwareCameraInstalled(getBaseContext())) {
+            if (ContextCompat.checkSelfPermission(getBaseContext(), Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+                if (ActivityCompat.shouldShowRequestPermissionRationale(HistoryScanActivity.this, Manifest.permission.CAMERA)) {
+                    new MaterialDialog.Builder(getBaseContext())
+                            .title(R.string.action_about)
+                            .content(R.string.permission_camera)
+                            .neutralText(R.string.txtOk)
+                            .onNeutral((dialog, which) -> ActivityCompat.requestPermissions(HistoryScanActivity.this, new String[]{Manifest.permission.CAMERA}, Utils.MY_PERMISSIONS_REQUEST_CAMERA))
+                            .show();
+                } else {
+                    ActivityCompat.requestPermissions(HistoryScanActivity.this, new String[]{Manifest.permission.CAMERA}, Utils.MY_PERMISSIONS_REQUEST_CAMERA);
+                }
+            } else {
+                Intent intent = new Intent(HistoryScanActivity.this, ScannerFragmentActivity.class);
+                startActivity(intent);
+            }
+        }
+
+    }
+
+    public void setInfo(TextView view) {
+
+        String info = "     Your viewed product history will be listed here.\n" +
```

Comment:
Are the spaces normal ?

Reply:
@teolemon To get the text centered justified , I had to put some spaces . I had already used android:gravity="center" but that did not solve the issue .

Reply:
@teolemon  I have solved it now . Thanks to suggestion from @Karljoones